### PR TITLE
xcute: Improve job's listing

### PR DESCRIPTION
**Pull request by AymericDu**
##### SUMMARY

Improve job's listing
- Change information visible when listing jobs
- Allow to add filters when listing jobs

##### ISSUE TYPE

 - Feature Pull Request

##### COMPONENT NAME

- xcute
- CLI

##### SDS VERSION

```
openio 5.7.1.dev9
```

##### ADDITIONAL INFORMATION

```
$ openio-admin xcute job list --date 2020-12-17 --status FINISHED --type rawx-decommission --lock rawx/127.0.0.1:6012
+----------------------------------+----------+-------------------+---------------------+----------+
| ID                               | Status   | Type              | Lock                | Progress |
+----------------------------------+----------+-------------------+---------------------+----------+
| 20201217152503729224-056ec779394 | FINISHED | rawx-decommission | rawx/127.0.0.1:6012 | 100.00%  |
+----------------------------------+----------+-------------------+---------------------+----------+
```

### CODE REVIEW

File: oio/cli/admin/xcute/job.py [hunk 1]
```diff
@@ -13,24 +13,87 @@
...
+            help='Filter jobs with the specified job date '
+                 '(%%Y-%%m-%%dT%%H:%%M:%%S)')
+        parser.add_argument(
+            '--status',
+            help='Filter jobs with the specified job status (%s)'
+                 % ', '.join(XcuteJobStatus.ALL))
+        parser.add_argument(
+            '--type',
+            help='Filter jobs with the specified job type (%s)'
+                 % ', '.join(JOB_TYPES.keys()))
```
Comment (murlock) [inline oio/cli/admin/xcute/job.py:44]: you should use `choices` instead

File: oio/xcute/common/backend.py [hunk 2]
```diff
@@ -545,6 +559,14 @@ def list_jobs(self, marker=None, limit=1000):
                     # The job can be deleted between two requests
                     continue
 
+                if job_status and job_info['job.status'] != job_status:
+                    continue
+                if job_type and job_info['job.type'] != job_type:
+                    continue
+                if job_lock and not fnmatchcase(
```
Comment (murlock) [inline oio/xcute/common/backend.py:566]: is it really necessary to use `fnmatchcase` ?

**Review by murlock — APPROVED** [review]

File: oio/xcute/common/backend.py (hunk 2 above)
Comment (AymericDu) [reply oio/xcute/common/backend.py:566]: With `fnmatchcase`, we allow listing a single service type (`rawx/*`), a single server (`*/127.0.0.1`), or even a single site.

Comment (murlock) [reply oio/xcute/common/backend.py:566]: :+1:

**Review by murlock — APPROVED** [review]

**Review by fvennetier — APPROVED** [review]

**Pull request merged**